packages/confirmation-modal/src/ConfirmationModal.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@leafygreen-ui/emotion';
+import Button, { Variant } from '@leafygreen-ui/button';
+import Modal from '@leafygreen-ui/modal';
+import { uiColors } from '@leafygreen-ui/palette';
+import TextInput from '@leafygreen-ui/text-input';
+
+const titleStyle = css`
+  color: ${uiColors.gray.dark2};
+  font-size: 24px;
+  font-weight: bold;
+  line-height: 25px;
+
+  margin-bottom: 10px;
+`;
+
+const baseModalStyle = css`
+  width: 600px;
+  padding: initial;
+`;
+
+const contentStyle = css`
+  color: ${uiColors.gray.dark1};
+  font-family: Akzidenz, ‘Helvetica Neue’, Helvetica, Arial, sans-serif;
+  font-size: 16px;
+  letter-spacing: 0;
+  line-height: 24px;
+
+  padding: 36px;
+`;
+
+const textEntryInputStyle = css`
+  width: 300px;
+  padding-top: 14px;
+`;
+
+const footerStyle = css`
+  &:before {
+    content: '';
+    display: block;
+    position: relative;
+    width: 100%;
+    border-top: 1px solid ${uiColors.gray.light2};
+  }
+`;
+
+const footerContentStyle = css`
+  display: flex;
+  justify-content: right;
+  flex-direction: row-reverse;
+  column-gap: 5px;
+  padding: 16px 24px;
+`;
+
+interface ConfirmationModalProps {
+  title: string;
+  children: React.ReactNode;
+  open?: boolean;
+  setOpen?: (open: boolean) => void;
+  className?: string;
+  primaryActionProps: ActionProps;
+  secondaryActionProps: ActionProps;
+  requireTextEntryConfirmation?: boolean;
+}
+
+interface ActionProps {
+  label: string;
+  onClick?: () => void;
+  variant?: Variant;
+}
+
+const ConfirmationModal = ({
+  children,
+  title,
+  primaryActionProps: { label: primaryActionLabel, ...primaryActionProps },
+  secondaryActionProps: {
+    label: secondaryActionLabel,
+    ...secondaryActionProps
+  },
+  requireTextEntryConfirmation = false,
+  ...modalProps
+}: ConfirmationModalProps) => {
+  const [primaryActionEnabled, setPrimaryActionEnabled] = useState(
+    !requireTextEntryConfirmation,
+  );
+
+  const textEntryConfirmation = useMemo(() => {
+    setPrimaryActionEnabled(!requireTextEntryConfirmation);
+
+    let textEntryConfirmation = null;
+
+    if (requireTextEntryConfirmation) {
```

Comment:
It might be a bit cleaner of an API if we use whether the app is passing `requiredText` to display this input. Is there a benefit to us requiring the consumer to pass two props to enable this feature?